src/librustc_mir/interpret/place.rs

```diff
@@ -455,7 +455,10 @@ where
     ) -> InterpResult<'tcx, MPlaceTy<'tcx, M::PointerTag>> {
         let len = base.len(self)?; // also asserts that we have a type where this makes sense
         let actual_to = if from_end {
-            assert!(from <= len - to);
+            if from + to > len {
```

Comment:
Can `from+to` overflow here?

Reply:
Even though the function is public, it's only called for `SubSlice`, which would require the user to specify more than 2^32 sub-patterns in a slice (`[x, y, z, subslice @ .., a, b, c]`),

so...

*probably not*, but there may be creative code generators out there